cpp/src/traversal/bfs_impl.cuh

```diff
@@ -222,7 +222,7 @@ void bfs(raft::handle_t const& handle,
         e_op.prev_visited_flags = prev_visited_flags.data();
       }
 
-      update_frontier_v_push_if_out_nbr(
+      update_frontier_from_outgoing_e_of_v(
```

Comment:
If we are using the sources/destinations nomenclature then this could be renamed to be
`update_frontier_from_destinations` because the frontier will only contain vertices that are destinations as opposed to sources.
Having said that `update_frontier_from_outgoing_edges` is fine too.

Reply:
What about `update_frontier_from_outgoing_e` to be more consistent with the rest? (we are using `e` to denote `edges` so I hope this abbreviation is not too confusing).

I initially used `e_of_v` instead of just `e` borrowing this convention from other primitives performing per vertex reduction, but here, we are not performing per vertex reduction. Starting from an input frontier, iterate over all the outgoing edges from the vertices in the frontier, then, compute a new frontier by filtering out destination vertices of the outgoing edges.

Something like `compute_new_frontier_from_outgoing_e_of_v_in_input_frontier` might be more descriptive (but too verbose) and I assume `update_frontier` part reasonably covers both `compute_new_frontier` and `of_v_in_input_frontier`.

Reply:
> What about update_frontier_from_outgoing_e to be more consistent with the rest?

Yes, this works